### PR TITLE
test: add tests for username case sensitive

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add integration tests for username case sensitivity.

## Testing

Integration tests.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/integration-tests/src/tests/api/admin-user.username-case.test.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  init: string[][] | Record<string, string> | URLSearchParams
+): Promise<{ headers: Headers; json: T }> => {
+  const response = await authedAdminApi.get('users', {
+    searchParams: new URLSearchParams(init),
+  });
+
+  return { headers: response.headers, json: (await response.json()) as T };
+};
+
+describe('admin console user management (username case insensitive)', () => {
```
Comment (gao-sun) [inline packages/integration-tests/src/tests/api/admin-user.username-case.test.ts:17]: these tests are for case-sensitive?

File: packages/integration-tests/src/tests/api/admin-user.username-case.test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const username = generateUsername();
+
+  it('should handle case sensitive in username', async () => {
```
Comment (gao-sun) [inline packages/integration-tests/src/tests/api/admin-user.username-case.test.ts:20]:
```suggestion
  it('should handle usernames case-sensitively', async () => {
```

File: packages/integration-tests/src/tests/api/admin-user.username-case.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    const user2 = await createUserByAdmin({ username: username.toUpperCase() });
+    await expectRejects(createUserByAdmin({ username: username.toUpperCase() }), {
+      code: 'user.username_already_in_use',
+      status: 422,
+    });
+    await deleteUser(user.id);
+    await deleteUser(user2.id);
+  });
+
+  it('should be able to search by username', async () => {
```
Comment (gao-sun) [inline packages/integration-tests/src/tests/api/admin-user.username-case.test.ts:31]:
```suggestion
  it('should be able to search by username case-insensitively', async () => {
```

File: packages/integration-tests/src/tests/api/admin-user.username-case.test.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    const user = await createUserByAdmin({ username: username.toLowerCase() });
+    const user2 = await createUserByAdmin({ username: username.toUpperCase() });
+    const { json } = await getUsers<User[]>([['search', `%${username.toLowerCase()}%`]]);
+    expect(json.length).toBe(2);
```
Comment (gao-sun) [inline packages/integration-tests/src/tests/api/admin-user.username-case.test.ts:35]:
```suggestion
    expect(json).toStrictEqual(...)
```

File: packages/integration-tests/src/tests/api/admin-user.username-case.test.ts
Comment (gao-sun) [inline packages/integration-tests/src/tests/api/admin-user.username-case.test.ts:1]: there's a typo in the pull description

**Review by darcyYe — APPROVED** [review]

**Pull request merged**